### DISCUSSION TITLE
[V5] Can the history plugin support behavior, layout, and plugin restoration?

### DISCUSSION THREAD

**Discussion by hongwei0417** [antvis/G6#6172] [Ideas]
The History plugin currently supports element recovery, but none element related operations are not supported.

**Comment by Aarebecca** [comment]
Some of the behaviors or plugins may not have been adapted yet, can you tell us which features you are having trouble using?

**Reply by hongwei0417** [reply]
I think can provide method to add custom commands to queue for custom actions not only limit node, edge and groups.
For example, i create a **custom plugin** that change the background image, and then i can restore the previous background by call a custom command.
Thanks 😄
